src/auth/interceptors/createRetryInterceptor.js

```diff
@@ -0,0 +1,63 @@
+import axios from 'axios';
+
+const createRetryInterceptor = (options = {}) => {
+  const {
+    httpClient = axios.create(),
+    // This default algorithm is a recreation of what is documented here
+    // https://cloud.google.com/storage/docs/exponential-backoff
+    getBackoffMs = (nthRetry, maximumBackoff = 32000) => {
+      const randomSecond = Math.random();
+      const exponentialBackoff = ((2 ** nthRetry) + randomSecond) * 1000;
+      return Math.min(Math.round(exponentialBackoff), maximumBackoff);
+    },
+    // By default only retry outbound request failures (not responses)
+    shouldRetry = error => !error.response && error.config,
+    // By default only retry 4 times, a per-request maxRetries can be
+    // specified in request config.
+    defaultMaxRetries = 4,
```

Comment:
Question: Should we limit this to two retries for now?

Reply:
I'd say 2 retries.

Reply:
Updated.